### PR TITLE
feat: enforce honest aggregate repository loading names

### LINKED ISSUE #491: Replace operation-labelled Project loads with honest access patterns
**Milestone:** Phase 13 Extraction Workflows — M1 (D1.0)
**PRD:** `docs/project/PRD/phase-13-extraction-workflows/PRD.md` §4 loading safety requirement
**Architecture:** `docs/project/PRD/phase-13-extraction-workflows/ARCH.md` §3.3
**Dogfooding:** Not applicable for this product-code deliverable
**Delivery plan:** `docs/project/PRD/phase-13-extraction-workflows/delivery.md` D1.0

## Problem

> Riviere extraction today is a sequence of manual CLI commands.

Current repository APIs overload `load(...)` and name access paths after later operations (`loadForEnrichment`, `loadWorkflow`). The role check cannot prevent more misleading public aggregate-loading names. This slice matters because D1.0 is an explicit dependency in the approved delivery sequence and an incomplete boundary would push design decisions into later implementation tickets.

## Solution

Enforce the `load` or `loadBy<AccessCriterion>` lexical rule, document semantic honesty, and migrate every current Project loading path without aliases or behaviour changes. This ticket delivers only this approved slice and leaves neighbouring capabilities to their named deliverables.

### Full architecture context

Read `docs/project/PRD/phase-13-extraction-workflows/ARCH.md` for the full approved architecture context and scope. This issue is not a replacement for `ARCH.md`.

### Target components, boundaries, and responsibilities

- Role enforcement selects public aggregate-returning repository methods.
- D1.0 names graph path, extraction config path, Draft Components path, and temporary Workflow name as the real criteria.
- Persistence methods and private assembly helpers remain unaffected.

### Code/config/flow from ARCH.md

```typescript
class RiviereProjectRepository {
  loadByGraphPath(graphPath: string): RiviereProject
  loadByExtractionConfigPath(input: ExtractionConfigLoadInput): RiviereProject
  loadByExtractionConfigAndDraftComponentsPaths(input: DraftComponentsLoadInput): RiviereProject
  loadByWorkflowName(input: WorkflowNameLoadInput): RiviereProject
}
```

```typescript
role('aggregate-repository', {
  allowedOutputs: ['aggregate'],
  outputMethodNameMatches: '^load(?:By[A-Z][A-Za-z0-9]*)?

### Role and location decisions

| Proposed Element | Kind | Role | Sublocation | Confidence | Notes |
| --- | --- | --- | --- | --- | --- |
| RoleConstraints.outputMethodNameMatches | data member | owned by `value-object` | role-enforcement domain `role-constraints.ts` | High | Carries the public output-method regex |
| BuiltRole.outputMethodNameMatches | data member | owned by `value-object` | role-enforcement domain `role-enforcement-builder.ts` | High | Projects the regex to plugin configuration |
| aggregate-repository plugin validation | plugin behaviour | existing role-enforcement boundary | role-enforcement plugin | High | Checks public aggregate-returning methods |
| RiviereProjectRepository | class | `aggregate-repository` | extract use-cases data-access/riviere-project | High | Migrates every current access path |

### Current-to-target replacement

Current flow: overloaded `load(string | LoadParameters)`, `loadForEnrichment(...)`, and `loadWorkflow(...)` all return `RiviereProject`.

Target flow: `loadByGraphPath(...)`, `loadByExtractionConfigPath(...)`, `loadByExtractionConfigAndDraftComponentsPaths(...)`, and temporary `loadByWorkflowName(...)`.

| Current path | Target path | Behaviour |
| --- | --- | --- |
| `load(graphFileLocation)` | `loadByGraphPath(graphFileLocation)` | Existing graph loading preserved |
| `load({ projectRoot, configPath, useTsConfig })` | `loadByExtractionConfigPath(...)` | Direct extraction loading preserved |
| `loadForEnrichment({ ..., draftComponentsPath })` | `loadByExtractionConfigAndDraftComponentsPaths(...)` | Draft Components-selected state preserved |
| `loadWorkflow({ projectRoot, workflowName })` | `loadByWorkflowName(...)` | Fixed name-addressed loading preserved until D1.5 |

The repository remains the `aggregate-repository`, inputs are real access criteria, and output remains one complete `RiviereProject`.

### Structural concerns and forbidden alternatives

- Do not constrain `save` or private helpers.
- Do not add canonical `load(workflowPath)` before D1.5.

### Design challenge instruction

> If implementation reveals the agreed design is impractical, incomplete, or sub-optimal, stop and push back. Do not silently implement a different design.

## What "done" looks like

### Product

- Every existing graph, extraction, enrichment, and Workflow command still loads the same Project state.
- Role diagnostics reject bare `loadBy`, `loadWorkflow`, `loadForEnrichment`, and `loadFromPersistedState`.

### Design

- Accepted names include every D1.0 method shown above.

- Old public names and compatibility aliases are absent.

### Quality

- Role tests cover exact accepted/rejected names, visibility, persistence methods, and repository call-site parity.

## Implementation guidelines

### Change areas

- `.riviere/role-definitions/aggregate-repository.md` and `.riviere/roles.ts`
- `packages/riviere-role-enforcement/` constraint model/plugin/tests
- `RiviereProjectRepository` and all callers/tests

### Relevant repository rules

- `docs/conventions/software-design.md` SD-001 (fail fast), SD-002/SD-003 (exhaustive type-safe states), SD-004 (dependency injection), SD-005 (intention-revealing names), and SD-021 (YAGNI) apply where this ticket touches code.
- `docs/conventions/testing.md` TS-001 through TS-004 require outcome-named tests with exact value assertions and one concept per test; TS-008/TS-014 require relevant boundary cases.

### Relevant architecture memories

- `project-memory/architecture/memories/repository-loading-must-load-full-aggregate.md`: repository criteria describe full aggregate state, not the operation that follows loading.
- `project-memory/architecture/memories/ddd-modelling-trade-offs-from-riviere-workflows.md`: Project owns empty-start rebuild meaning and Workflow owns the stage lifecycle; repositories and use cases do not become the engine.
- `project-memory/architecture/memories/rejected-workflow-use-case-dumping-case-study.md`: do not dump stage progression or low-level collaborators into a command use case.

### Lint, role-check, and test expectations

Use role and lint failures to improve the design; do not add ignores, exclusions, assertions, optional production dependencies, or coverage workarounds. Preserve exhaustive unions and avoid the TypeScript `in` operator.

## Dependencies

- None.

## How to verify

- Run focused role-enforcement and extraction-use-case tests; exact naming and call-site parity assertions pass.
- Run `pnpm role-check`; no aggregate-repository loading-name violation remains.
- Run `pnpm verify`; the repository is green without exceptions.

## Things this ticket must not do

- Do not accept operation-labelled `loadByEnrichment`.
- Do not constrain `save` or private helpers.
- Do not add canonical `load(workflowPath)` before D1.5.
,
})
```

### Role and location decisions

| Proposed Element | Kind | Role | Sublocation | Confidence | Notes |
| --- | --- | --- | --- | --- | --- |
| RoleConstraints.outputMethodNameMatches | data member | owned by `value-object` | role-enforcement domain `role-constraints.ts` | High | Carries the public output-method regex |
| BuiltRole.outputMethodNameMatches | data member | owned by `value-object` | role-enforcement domain `role-enforcement-builder.ts` | High | Projects the regex to plugin configuration |
| aggregate-repository plugin validation | plugin behaviour | existing role-enforcement boundary | role-enforcement plugin | High | Checks public aggregate-returning methods |
| RiviereProjectRepository | class | `aggregate-repository` | extract use-cases data-access/riviere-project | High | Migrates every current access path |

### Current-to-target replacement

Current flow: overloaded `load(string | LoadParameters)`, `loadForEnrichment(...)`, and `loadWorkflow(...)` all return `RiviereProject`.

Target flow: `loadByGraphPath(...)`, `loadByExtractionConfigPath(...)`, `loadByExtractionConfigAndDraftComponentsPaths(...)`, and temporary `loadByWorkflowName(...)`.

| Current path | Target path | Behaviour |
| --- | --- | --- |
| `load(graphFileLocation)` | `loadByGraphPath(graphFileLocation)` | Existing graph loading preserved |
| `load({ projectRoot, configPath, useTsConfig })` | `loadByExtractionConfigPath(...)` | Direct extraction loading preserved |
| `loadForEnrichment({ ..., draftComponentsPath })` | `loadByExtractionConfigAndDraftComponentsPaths(...)` | Draft Components-selected state preserved |
| `loadWorkflow({ projectRoot, workflowName })` | `loadByWorkflowName(...)` | Fixed name-addressed loading preserved until D1.5 |

The repository remains the `aggregate-repository`, inputs are real access criteria, and output remains one complete `RiviereProject`.

### Structural concerns and forbidden alternatives

- Do not accept operation-labelled `loadByEnrichment`.
- Do not constrain `save` or private helpers.
- Do not add canonical `load(workflowPath)` before D1.5.

### Design challenge instruction

> If implementation reveals the agreed design is impractical, incomplete, or sub-optimal, stop and push back. Do not silently implement a different design.

### Product

- Every existing graph, extraction, enrichment, and Workflow command still loads the same Project state.
- Role diagnostics reject bare `loadBy`, `loadWorkflow`, `loadForEnrichment`, and `loadFromPersistedState`.

### Design

- Accepted names include every D1.0 method shown above.
- `loadByEnrichment` is rejected semantically even though it satisfies the lexical prefix.
- Old public names and compatibility aliases are absent.

### Quality

- Role tests cover exact accepted/rejected names, visibility, persistence methods, and repository call-site parity.

### Change areas

- `.riviere/role-definitions/aggregate-repository.md` and `.riviere/roles.ts`
- `packages/riviere-role-enforcement/` constraint model/plugin/tests
- `RiviereProjectRepository` and all callers/tests

### Relevant repository rules

- `docs/conventions/software-design.md` SD-001 (fail fast), SD-002/SD-003 (exhaustive type-safe states), SD-004 (dependency injection), SD-005 (intention-revealing names), and SD-021 (YAGNI) apply where this ticket touches code.
- `docs/conventions/testing.md` TS-001 through TS-004 require outcome-named tests with exact value assertions and one concept per test; TS-008/TS-014 require relevant boundary cases.

### Relevant architecture memories

- `project-memory/architecture/memories/repository-loading-must-load-full-aggregate.md`: repository criteria describe full aggregate state, not the operation that follows loading.
- `project-memory/architecture/memories/ddd-modelling-trade-offs-from-riviere-workflows.md`: Project owns empty-start rebuild meaning and Workflow owns the stage lifecycle; repositories and use cases do not become the engine.
- `project-memory/architecture/memories/rejected-workflow-use-case-dumping-case-study.md`: do not dump stage progression or low-level collaborators into a command use case.

### Lint, role-check, and test expectations

Use role and lint failures to improve the design; do not add ignores, exclusions, assertions, optional production dependencies, or coverage workarounds. Preserve exhaustive unions and avoid the TypeScript `in` operator.

## Dependencies

- None.

## How to verify

- Run focused role-enforcement and extraction-use-case tests; exact naming and call-site parity assertions pass.
- Run `pnpm role-check`; no aggregate-repository loading-name violation remains.
- Run `pnpm verify`; the repository is green without exceptions.

## Things this ticket must not do

- Do not accept operation-labelled `loadByEnrichment`.
- Do not constrain `save` or private helpers.
- Do not add canonical `load(workflowPath)` before D1.5.

### PR DESCRIPTION
**Pull request by NTCoding**
## Description

Enforces a lexical naming rule for public aggregate-returning repository callables and renames the Project repository's dual-criterion loading path to name both real access criteria.

## Linked Issue

Closes #491

## What Problem Does This PR Solve?

Aggregate repositories could expose operation-labelled public loading names that obscure the actual criterion used to retrieve the complete aggregate.

## Acceptance Criteria

- Public aggregate-returning repository callables use load or loadBy<AccessCriterion> names.
- Invalid dynamic and numeric callable names are rejected.
- Project repository loading call sites preserve behavior and name both extraction configuration and Draft Components paths when both are required.

## Key Changes

- Added aggregate-repository callable-name enforcement and AST regression coverage.
- Scoped the rule to aggregate-returning public callables.
- Renamed loadByDraftComponentsPath to loadByExtractionConfigAndDraftComponentsPaths across implementation, tests, and Phase 13 records.

## Notable Architectural Changes / Impact

The aggregate-repository role now makes lexical loading-name constraints executable while leaving semantic suitability to architecture review.

## Validation

- pnpm verify (passed)

## Notes

No compatibility aliases were added; the renamed API identifies its actual loading criteria.

## Summary by CodeRabbit

* **Changes**
  * Standardized project-loading method names to clearly indicate the access criteria used, including graph path, extraction configuration, draft components, and workflow name.
  * Loading by a plain graph-path argument through the previous generic method is no longer supported; use the dedicated graph-path method instead.
  * Added validation to ensure aggregate repository loading methods follow approved naming patterns.

* **Documentation**
  * Updated extraction workflow documentation and acceptance criteria to use the revised loading method names.

* **Tests**
  * Expanded coverage for loading-name validation and supported public member shapes.

### CODE REVIEW

File: packages/riviere-role-enforcement/domain-model/src/domain/role-constraints.ts [hunk 1]
```diff
@@ -119,6 +121,24 @@
...
   }
 }
+
+function validateRegularExpression(pattern: string | undefined, constraint: string): void {
+  if (pattern === undefined) {
+    return
+  }
+
+  try {
+    new RegExp(pattern)
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply packages/riviere-role-enforcement/domain-model/src/domain/role-constraints.ts:138]: This is a false positive. `pattern` comes from trusted executable TypeScript role configuration; anyone able to control it can already execute arbitrary code through that configuration. Parsing validates syntax but does not introduce a separate untrusted-input boundary.

File: packages/riviere-role-enforcement/domain-model/role-enforcement-plugin.mjs [hunk 2]
```diff
@@ -2038,19 +2038,179 @@
...
+
+          const outputRoles = readOutputTypeRoles(callableMember.functionNode.returnType, filename)
+          if (
+            outputRoles === null ||
+            !outputRoles.some((outputRole) => role.allowedOutputs.includes(outputRole))
+          ) {
+            return
+          }
+
+          if (new RegExp(role.outputMethodNameMatches).test(callableMember.methodName)) {
```
[the remark this answers is not in the document]
Comment (NTCoding) [reply packages/riviere-role-enforcement/domain-model/role-enforcement-plugin.mjs:2083]: This is a false positive. `outputMethodNameMatches` comes from trusted executable TypeScript role configuration; anyone able to control it can already execute arbitrary code through that configuration. The regex is tested only against source member names and adds no separate untrusted-input boundary.

**Comment by NTCoding** [comment]
@coderabbitai review
